### PR TITLE
stella-cv-fbow: new recipe

### PR DESCRIPTION
**Pull request by valgur**
Adds a stella-cv fork of FBoW: https://github.com/stella-cv/FBoW

FBoW (Fast Bag of Words) is an extremely optimized version of the [DBoW2](https://github.com/dorian3d/DBoW2)/[DBoW3](https://github.com/rmsalinas/DBow3) libraries.
The library is highly optimized to speed up the Bag of Words creation using AVX,SSE and MMX instructions.

Required for packaging of https://github.com/stella-cv/stella_vslam

### CODE REVIEW

File: recipes/stella-cv-fbow/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        # https://github.com/stella-cv/FBoW/blob/master/include/fbow/vocabulary.h#L35
+        self.requires("opencv/4.9.0", transitive_headers=True, transitive_libs=True)
+        self.requires("llvm-openmp/17.0.6")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, 11)
```
Comment (RubenRBS) [inline recipes/stella-cv-fbow/all/conanfile.py:69]: Shouldn't we also remove https://github.com/stella-cv/FBoW/blob/c6e3c29e3332a0b0834021797e2aa4e8eb66a3c1/CMakeLists.txt#L112?

**Review by RubenRBS — COMMENTED** [review]
Thanks! Minor comment about the cmake cxx standard variable, otherwise looks good :)

File: recipes/stella-cv-fbow/all/conanfile.py (hunk 1 above)
Comment (valgur) [reply recipes/stella-cv-fbow/all/conanfile.py:69]: Done.

**Review by RubenRBS — DISMISSED** [review]

File: recipes/stella-cv-fbow/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.15)
+project(test_package LANGUAGES C)
+project(test_package LANGUAGES CXX)
```
Comment (toge) [inline recipes/stella-cv-fbow/all/test_package/CMakeLists.txt:3]:
```suggestion
project(test_package LANGUAGES CXX)
```

There seems to be no c file.

**Review by jwillikers — APPROVED** [review]

**Review by RubenRBS — DISMISSED** [review]

**Comment by perseoGI** [comment]
Hi @valgur, is this PR still in draft? Or can we merge?

**Review by perseoGI — APPROVED** [review]
LGTM!

File: recipes/stella-cv-fbow/all/test_package/test_package.cpp [hunk 3]
```diff
@@ -1,5 +1,10 @@
 #include <fbow/fbow.h>
+#include <fbow/cpu.h>
+#include <iostream>
 
 int main() {
-    fbow::cpu().detect_host();
+    auto features = fbow::cpu();
+    features.detect_host();
+    std::cout << "CPU Vendor String: " << fbow::cpu::get_vendor_string() << '\n';
```
Comment (jwillikers) [inline recipes/stella-cv-fbow/all/test_package/test_package.cpp:8]: It warms my heard to see `<< '\n'` and not `std::endl`.

**Review by jwillikers — APPROVED** [review]

File: recipes/stella-cv-fbow/all/test_package/test_package.cpp (hunk 3 above)
Comment (perseoGI) [reply recipes/stella-cv-fbow/all/test_package/test_package.cpp:8]: ❤️

**Pull request merged**